src/pru-hardware-interface.h

```diff
@@ -0,0 +1,60 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * (c) 2013, 2016 Henner Zeller <h.zeller@acm.org>
+ *
+ * This file is part of BeagleG. http://github.com/hzeller/beagleg
+ *
+ * BeagleG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BeagleG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BeagleG.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef BEAGLEG_PRU_HARDWARE_INTERFACE_
+#define BEAGLEG_PRU_HARDWARE_INTERFACE_
+
+#include <cstddef>
+#include <pruss_intc_mapping.h>
+
+// Target PRU
+#define PRU_NUM 0
+
+#define SEGMENT_EXECUTED PRU_EVTOUT0
+
+// Pru hardware controls
+class PruHardwareInterface {
+public:
+  virtual ~PruHardwareInterface() {}
+
+  // Initialize the hardware, load the binary, interrupts, etc...
+  virtual int Init() = 0;
+
+  // Retrieve the pointer of the pru mapping and initialize the memory.
+  virtual int Allocate(void **pru_mmap, const size_t size) = 0;
+
+  // Enable the PRU
+  virtual int Execute() = 0;
+
+  // Wait for a beagleg-mapped event
+  virtual int WaitEvent(const unsigned int eventnum) = 0;
```

Comment:
At this point: do we need more than one event type ? If not, I'd just leave this out and leave it an implementation detail of the implementor how that is done internally.

Because otherwise, the pru-motion-queue needs to include the PRU headers to know special symbols such as PRU_EVTOUT_0 ... which  means we actually lost the abstraction.